src/ReactiveUI/ReactiveNotifyPropertyChangedMixin.cs

```diff
@@ -27,22 +22,25 @@ static ReactiveNotifyPropertyChangedMixin()
         /// IObservedChange) guarantees that the Value property of
         /// the IObservedChange is set.
         /// </summary>
+        /// <param name="this">The source object to observe properties of</param>
         /// <param name="property">An Expression representing the property (i.e.
         /// 'x => x.SomeProperty.SomeOtherProperty'</param>
         /// <param name="beforeChange">If True, the Observable will notify
         /// immediately before a property is going to change.</param>
+        /// <param name="skipInitial">If true, the Observable will not notify 
+        /// with the initial value</param>
         /// <returns>An Observable representing the property change
         /// notifications for the given property.</returns>
         public static IObservable<IObservedChange<TSender, TValue>> ObservableForProperty<TSender, TValue>(
-                this TSender This,
+                this TSender @this,
                 Expression<Func<TSender, TValue>> property,
                 bool beforeChange = false,
                 bool skipInitial = true)
         {
-            if (This == null) {
-                throw new ArgumentNullException("Sender");
+            if (@this == null) {
+                throw new ArgumentNullException(nameof(@this));
```

Comment:
This is potentially a breaking change so happy to remove and just log instead...